armi/bookkeeping/db/jaggedArray.py

```diff
@@ -0,0 +1,191 @@
+# Copyright 2024 TerraPower, LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+Tooling to help flatten jagged (non rectangular) data into rectangular arrays.
+
+The goal here is to support jagged data for NumPy arrays to be written into the ARMI databases.
+"""
+
+from typing import List, Optional
+
+import numpy as np
+
+from armi import runLog
+
+
+class JaggedArray:
```

Comment:
I suppose I wouldn't _hate_ having unit tests directly for this class.

Reply:
Fair enough, I'll work on that

Reply:
Okay, thanks.

Other than that, I think this PR is ready for prime time. 

Holler when you're ready, and I'm happy to get this merge party started.

Reply:
Okay, I added some unit tests. Mostly testing the ability to round-trip data to the HDF5 file and back. I think it may have improved the coverage a bit.